app/src/main/java/com/teamagam/gimelgimel/app/network/services/message_polling/MessageLocalBroadcaster.java

```diff
@@ -0,0 +1,33 @@
+package com.teamagam.gimelgimel.app.network.services.message_polling;
+
+import android.content.Context;
+
+import com.teamagam.gimelgimel.app.model.ViewsModels.Message;
+import com.teamagam.gimelgimel.app.model.ViewsModels.MessageBroadcastReceiver;
+import com.teamagam.gimelgimel.app.utils.NetworkUtil;
+
+/**
+ * Wraps Local-broadcasting functionality
+ */
+public class MessageLocalBroadcaster implements IMessageBroadcaster {
+
+    private Context mContext;
+
+    public MessageLocalBroadcaster(Context context) {
+        mContext = context;
+    }
+
+    @Override
+    public void broadcast(Message message) {
+        if (message == null) {
+            throw new IllegalArgumentException("Message cannot be null");
+        }
+
+        //Do not broadcast messages from self
+        if (message.getSenderId().equals(NetworkUtil.getMac())) {
+            return;
+        }
+
```

Comment:
I think that DEBUG severity is too high for a single message. Ill log it as Verbose and change the number of incoming messages to DEBUG